superset/jinja_context.py

```diff
@@ -122,12 +78,52 @@ def filter_values(column: str, default: Optional[str] = None) -> List[str]:
     return []
 
 
-class CacheKeyWrapper:  # pylint: disable=too-few-public-methods
+class CacheKey:  # pylint: disable=too-few-public-methods
     """ Dummy class that exposes a method used to store additional values used in
      calculation of query object cache keys"""
 
+    # Regular expression for detecting the presence of templated methods which could
+    # be added to the cache key.
+    regex = re.compile(
+        r"\{\{.*("
+        r"current_user_id\(.*\)|"
+        r"current_username\(.*\)|"
+        r"cache_key_wrapper\(.*\)|"
+        r"url_param\(.*\)"
+        r").*\}\}"
+    )
+
     def __init__(self, extra_cache_keys: Optional[List[Any]] = None):
-        self.extra_cache_keys = extra_cache_keys
+        self.extra_cache_keys = extra_cache_keys or []
+
+    def current_user_id(self, add_to_cache_keys: bool = True) -> Optional[int]:
```

Comment:
I think the references to the moved methods need to be updated in `docs/sqllab.rst`.